packages/terra-signature/src/Signature.jsx

```diff
@@ -0,0 +1,215 @@
+import styles from './Signature.scss';
+
+const LineWidths = {
+  EXTRA_FINE: 1,
+  FINE: 2,
+  MEDIUM: 4,
+  HEAVY: 6,
+};
+const propTypes = {
+  /**
+   * The line width to use when drawing the signature on the canvas.
+   * One of LineWidths.EXTRA_FINE, LineWidths.FINE, LineWidths.MEDIUM, LineWidths.HEAVY.
+   */
+  lineWidth: PropTypes.oneOf([LineWidths.EXTRA_FINE, LineWidths.FINE, LineWidths.MEDIUM, LineWidths.HEAVY]),
+  /**
+  * Line segments that define signature.
+  */
+  lineSegments: PropTypes.array,
+  /**
+   * A callback function to execute when a line segment is drawn. The first parameter is the event, the
+   * second parameter is all the line segments, and the last parameter is the most recent line segment drawn.
+   */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  lineWidth: LineWidths.FINE,
+  lineSegments: undefined,
+  onChange: undefined,
+};
+
+class Signature extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = { lineSegments: props.lineSegments, lineWidth: props.lineWidth };
+
+    this.mouseDown = this.mouseDown.bind(this);
+    this.mouseUp = this.mouseUp.bind(this);
+    this.mouseXY = this.mouseXY.bind(this);
+    this.addLine = this.addLine.bind(this);
+    this.draw = this.draw.bind(this);
+    this.drawSignature = this.drawSignature.bind(this);
+    this.clearSignature = this.clearSignature.bind(this);
+    this.updateDimensions = this.updateDimensions.bind(this);
+  }
+
+  componentDidMount() {
+    const canvas = this.node;
+
+    if ('ontouchstart' in document.documentElement) {
+      this.node.addEventListener('touchstart', this.mouseDown, false);
+      this.node.addEventListener('touchmove', this.mouseXY, true);
+      this.node.addEventListener('touchend', this.mouseUp, false);
+      document.body.addEventListener('touchcancel', this.mouseUp, false);
+    } else {
+      this.node.addEventListener('mousedown', this.mouseDown);
+      this.node.addEventListener('mousemove', this.mouseXY);
+      document.body.addEventListener('mouseup', this.mouseUp);
+    }
+
+    this.node.width = canvas.getBoundingClientRect().width;
+    this.node.height = canvas.getBoundingClientRect().height;
+
+    const context = this.node.getContext('2d');
+
+    context.lineWidth = this.props.lineWidth;
+
+    const style = window.getComputedStyle(this.node);
+    const color = style.getPropertyValue('color');
+    context.strokeStyle = color;
+
+    this.drawSignature(this.props.lineSegments, this.props.lineWidth);
+
+    window.addEventListener('resize', () => this.updateDimensions());
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if ((this.props.lineSegments !== nextProps.lineSegments) ||
+      (this.props.lineWidth !== nextProps.lineWidth)) {
+      this.setState({ lineSegments: nextProps.lineSegments, lineWidth: nextProps.lineWidth });
+      this.drawSignature(nextProps.lineSegments, nextProps.lineWidth);
+    }
+  }
+
+  componentWillUnmount() {
```

Comment:
Will we want to unmount 'onTouchStart' event listeners?

Reply:
fixed: https://github.com/cerner/terra-core/pull/769/commits/3a8c2f78444b155b7b6ec315126022c03c9bb5d9